include/toml++/impl/path.h

```diff
@@ -23,20 +23,59 @@ TOML_NAMESPACE_START
 	using path_component_value = std::variant<std::string, size_t>;
 
 	/// \brief Represents a single component of a complete 'TOML-path': either a key or an array index
-	struct path_component
+	struct TOML_EXPORTED_CLASS path_component
 	{
-		path_component_value value;
-		path_component_type type;
+		friend class path;
 
 	  private:
 		/// \cond
 
+	  	path_component_value value_;
+		path_component_type type_;
+
 		TOML_PURE_GETTER
 		TOML_EXPORTED_STATIC_FUNCTION
 		static bool TOML_CALLCONV equal(const path_component&, const path_component&) noexcept;
 
 		/// \endcond
 	  public:
+		/// \brief	Default constructor.
+		TOML_NODISCARD_CTOR
+		path_component() noexcept = default;
+
+		/// \brief	Constructor for a path component that is an array index
+		TOML_NODISCARD_CTOR
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path_component(size_t index);
```

Comment:
```suggestion
		TOML_NODISCARD_CTOR
		TOML_EXPORTED_MEMBER_FUNCTION
		path_component(size_t index) noexcept;
```

I'd be very surprised if the `std::variant` constructor for the `size_t` instantiation was capable of throwing